Flutter/datagrid/column-drag-and-drop.md

```diff
@@ -0,0 +1,338 @@
+---
+layout: post
+title: Column drag and drop in Flutter DataGrid | DataTable | Syncfusion
+description: Learn all about how to drag and drop the columns in Syncfusion Flutter DataGrid (SfDataGrid) widget and more here.
+platform: flutter
+control: SfDataGrid
+documentation: ug
+--- 
+
+# Column drag and drop in Flutter DataGrid (SfDataGrid)
+
+The SfDataGrid allows column header dragging and dropping by setting the `SfDataGrid.allowColumnsDragging` property to `true` and returning `true` from the `SfDataGrid.onColumnDragging` callback. During the column dragging process, a drag feedback widget is displayed. By utilizing the `SfDataGrid.onColumnDragging` event, you can handle drag and drop operations according to your requirements.
+
+The DataGrid provides the rearranged index of the dragged column, indicating its new position after being dropped. Inside the `onColumnDragging` callback, you can utilize this index to reorder the columns according to the desired position. This allows you to handle the column reordering directly within the callback at the sample level.
+
+{% tabs %}
+{% highlight Dart %} 
+
+class MyHomePageState extends State<MyHomePage> {
+  List<Employee> employees = <Employee>[];
+  late List<GridColumn> columns;
+  late EmployeeDataSource employeeDataSource;
+
+  @override
+  void initState() {
+    super.initState();
+    columns = getColumns;
+    employees = getEmployeeData();
+    employeeDataSource =
+        EmployeeDataSource(employees: employees, columns: columns);
+  }
+
+  List<GridColumn> get getColumns {
+    return <GridColumn>[
+      GridColumn(
+          columnName: 'id',
+          label: Container(
+              padding: const EdgeInsets.all(16.0),
+              alignment: Alignment.center,
+              child: const Text(
+                'ID',
+              ))),
+      GridColumn(
+          columnName: 'name',
+          label: Container(
+              padding: const EdgeInsets.all(8.0),
+              alignment: Alignment.center,
+              child: const Text('Name'))),
+      GridColumn(
+          columnName: 'designation',
+          label: Container(
+              padding: const EdgeInsets.all(8.0),
+              alignment: Alignment.center,
+              child: const Text(
+                'Designation',
+                overflow: TextOverflow.ellipsis,
+              ))),
+      GridColumn(
+          columnName: 'salary',
+          label: Container(
+              padding: const EdgeInsets.all(8.0),
+              alignment: Alignment.center,
+              child: const Text('Salary'))),
+    ];
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
build method and DataGridSource changes will be enough. Link the GitHub sample for the column drag-and-drop demo.